### PR TITLE
Warn about derived quantities in non cartesian

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
## Proposed changes

This PR adds a warning when users employ `SurfaceFlux` with a non-cartesian mesh (eg. an axisymmetric 2D mesh).

Indeed, the flux (example here for particles) expression is

$$
J_\mathrm{total} = \int J \cdot n \ dS
$$

where $n$ is the normal vector and

$$
J = - D \nabla c
$$

In cartesian coordinates, the element $dS$ is $dS = dx \ dy $, assuming it is $y$ independent $dS = y \ dx$.
And this is the default behaviour in fenics.

However in cylindrical $(r, \theta, z)$, the surface element is $dS = r \ dr \ d\theta$, assuming it is axisymmetric ($\theta$ independent), then $dS =\theta \ r \ dr$

The flux becomes:

$$
J_\mathrm{total} = \int J \cdot n \ dS = \theta \int r \ J \cdot n \ dr
$$

which is NOT the default behaviour of `SurfaceFlux`.
While we wait for a native support of fluxes in cylindrical coordinates to be implemented, users should at least be warn that this may be an issue.

The same goes for `TotalVolume` , `AverageVolume` , etc.

Thanks @WeaverC20 for helping spotting this

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
@jhdark can we merge this before #689 ?

File: festim/generic_simulation.py [hunk 1]
```diff
@@ -270,6 +270,26 @@ def initialise(self):
 
         self.h_transport_problem.initialise(self.mesh, self.materials, self.dt)
 
+        # raise warning if SurfaceFlux is used with non-cartesian meshes
+        for export in self.exports.exports:
```
Comment (KulaginVladimir) [inline festim/generic_simulation.py:274]: just to remind, `.exports` will be deprecated

Comment (RemDelaporteMathurin) [reply festim/generic_simulation.py:274]: Yes I probably have some merge conflicts I need to take care of

**Review by jhdark — APPROVED** [review]
Looks good to me!

**Pull request merged**